### PR TITLE
multitenant: append `WITH REPLICATION STATUS` columns to `SHOW TENANT` columns

### PR DESCRIPTION
**Pull request by ecwall**
Informs #87851

`WITH CAPABILITIES` will be adding another set of columns so append relevant columns instead of creating all possible column sets ahead of time.

Release note: None

### CODE REVIEW

File: pkg/sql/show_tenant.go [hunk 1]
```diff
@@ -73,10 +73,10 @@ func (p *planner) ShowTenant(ctx context.Context, n *tree.ShowTenant) (planNode,
...
 		withReplication: n.WithReplication,
 	}
+
+	node.columns = colinfo.TenantColumns
 	if n.WithReplication {
-		node.columns = colinfo.TenantColumnsWithReplication
-	} else {
-		node.columns = colinfo.TenantColumns
+		node.columns = append(node.columns, colinfo.TenantColumnsWithReplication...)
 	}
```
Comment (ecwall) [inline pkg/sql/show_tenant.go:81]: The capabilities PR will add
```
	if n.WithCapabilities {
		node.columns = append(node.columns, colinfo.TenantColumnsWithCapabilities...)
	}
```

File: pkg/sql/show_tenant.go [hunk 2]
```diff
@@ -234,61 +234,59 @@ func (n *showTenantNode) Next(params runParams) (bool, error) {
...
+		result = append(result,
+			sourceTenantName,
+			sourceClusterUri,
+			replicationJobId,
+			replicatedTimestamp,
+			retainedTimestamp,
+			cutoverTimestamp,
+		)
 	}
+
```
Comment (ecwall) [inline pkg/sql/show_tenant.go:288]: The capabilities PR will add
```
  if n.WithReplication {
    result = append(result, ...)
  }
```

**Review by knz — APPROVED** [review]
Nice.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/95526#-:-NMAyM2pA4IwDyygSihI:b-83sjo4)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @rafiss)

**Comment by ecwall** [comment]
bors r=knz

**Pull request merged**